### DISCUSSION TITLE
Force changing admin account

### DISCUSSION THREAD

**Discussion by Xevalous** [pocketbase/pocketbase#1114] [Q&A] [answered]
Basicly i kinda forgot my admin account in local pb when going to log in & i dont wanna my db gone, i know its dumb, i have tried to reset password from login page, but there is none email sended

soo, is there a other method to reset admin pass?

**Comment by ganigeorgiev** [comment, marked as answer]
In the future there will be a console command(s) to interact with the Admin APIs, but for now if you want to reset your password you can directly update the _admins table entry in your database:

```
sqlite3 pb_data/data.db 'UPDATE _admins SET passwordHash="$2a$10$xWbkSks7OiTmrri2DU6QSe6OAekcdqkOrkcI1wi4uiv9SU2HTlHoq" where email="test@example.com"'
```

The above will set a `123456` password for the admin with email `test@example.com` (you can remove the `where ...` part if you don't remember your email).

_**Update:** I've inversed the quotes in the above command to prevent the command line to interpret the password hash as env variable._

**Reply by grabbels** [reply]
Is it correct that this does not work anymore? The command outputs no error, but the 123456 password does not work when trying to log in after.

**Reply by ganigeorgiev** [reply]
@grabbels It should work just fine. Make sure that you are using the correct email in the `where` clause. If you have only 1 admin and don't know its email, you can safely remove the `where ...` part.

**Reply by grabbels** [reply]
I did confirm the change and sqlite lists the passwordHash as identical to the one you mentioned, still not working. Email is known and correct. I've restarted the pocketbase server as an extra measure but no change in the situation unfortunately.

**Reply by ganigeorgiev** [reply]
@grabbels I've just tested it locally and it works fine for me. You can try without the email (maybe the account was created with a typo?):

```sh
sqlite3 pb_data/data.db 'UPDATE _admins SET passwordHash="$2a$10$xWbkSks7OiTmrri2DU6QSe6OAekcdqkOrkcI1wi4uiv9SU2HTlHoq"'
```

To confirm that the password hash was changed, you can run:
```sh
sqlite3 pb_data/data.db 'select passwordHash from _admins'
```

**Reply by grabbels** [reply]
I checked everything. My email is correct and the passwordHash has been changed correctly. Still not getting into the backend: "Invalid login credentials.".
The database currently contains no critical data whatesoever so I could just as well reinstall pocketbase alltogether, but I just want to make sure this is not some kind of unexpected behavior that might be rooted in a bug of some sort.

**Reply by ganigeorgiev** [reply]
@grabbels Alternatively, you can try to delete all admins and create a new one from the Admin UI:

```sh
# delete all admins and then go to http://127.0.0.1:8090/_/?installer
sqlite3 pb_data/data.db 'delete from _admins'
```

**Reply by ganigeorgiev** [reply]
> @grabbels The database currently contains no critical data whatesoever so I could just as well reinstall pocketbase alltogether, but I just want to make sure this is not some kind of unexpected behavior that might be rooted in a bug of some sort.

As mentioned, I'm not able to reproduce it. There is something strange happening in your environment, but there are not enough details to help you debug it.

**Reply by grabbels** [reply]
I understand,  is there any information I could dig up for you that might help? 

`# delete all admins and then go to http://127.0.0.1:8090/_/?installer
sqlite3 pb_data/data.db 'delete from _admins'`
http://127.0.0.1:8090/_/?installer redirects to http://127.0.0.1:8090/_/#/login. I'm kind of at a loss here, everything seems to be properly working except for these weird things happening. I haven't had any other problems with pocketbase in the few months I've been testing it now so I'm quite confused. I'll go for a full reinstall I suppose as I of course understand there's not much you can do to help me in this situation.

**Reply by ganigeorgiev** [reply]
@grabbels If the installer redirects to login then this means that the above command didn't delete your admins, aka. you are not making changes to the database that the PocketBase executable use.

Are you sure that your PocketBase executable uses the `pb_data` directory you are modifying?

You can specify an explicit `pb_data` directory using the `--dir` flag:

```js
./pocketbase serve --dir="/path/to/pb_data"
```

**Reply by grabbels** [reply]
Alright, that did it. Somehow it defaulted to using pb_data located in /root, which is very confusing to me. Not sure how that happened.
Thank you so much for your help and I'm sorry to take up your time like this!

**Reply by ganigeorgiev** [reply]
@grabbels

> Somehow it defaulted to using pb_data located in /root

This is because by default the executable uses (creates) the `pb_data` directory in the same parent directory of the executable, aka. most likely your executable is located inside `/root`.
